app/policies/appeal_decision_issues_policy.rb

```
@@ -20,7 +20,11 @@ def restricted_decision_issues
     if @user.vso_employee?
       visible_issues = @appeal.decision_issues.select do |issue|
         # VSO users should not be able to see decision issues until the issue decision date
-        Time.now.utc > issue.caseflow_decision_date
+        begin
+          Time.now.utc > issue.approx_decision_date
+        rescue StandardError
+          false
```

This seems like a good  way to handle things, but I think running it by Sally or Yang would be a good idea as a sanity check regarding issue dates

---

Judging from the safe-navigation operator on [this line](https://github.com/department-of-veterans-affairs/caseflow/blob/15ed93756eb6b9437b2f3bdc514ae43f99ec97bd/app/models/concerns/issue_updater.rb#L50) I would guess that nil decision dates are possible (and the source checks out; the Appeal from that Bat Team issue does indeed have a decision issue but no decision document).

In the context of this policy, interpreting nil to mean infinitely far in the future is a reasonable extrapolation.

My only suggestion is to stay away from the blanket `rescue StandardError` which is normally a code smell, and a pretty accurate one here: you only want to return false if `approx_decision_date` is nil, but if anything else goes wrong you'd still want to raise an exception.

---

TIL: if you wanted to get real cute, you could do `> (issue.approx_decision_date || DateTime::Infinity.new)`

---

Thanks for the input, @nanotone!  I also reached out to Sally via slack, and she confirmed that decision issues don't always have a decision date.